client/web/app/elements/lecture/editor.html

```diff
@@ -1,62 +1,242 @@
+                        "src/spring/output-2.png",
+                        "src/spring/output-3.png",
+                        "src/spring/output-4.png",
+                        "src/spring/output-5.png",
+                        "src/spring/output-6.png",
+                        "src/spring/output-7.png",
+                        "src/spring/output-8.png",
+                        "src/spring/output-9.png",
+                        "src/spring/output-10.png",
+                        "src/spring/output-11.png"
+                    ]
+                }
+            },
+            observers: [
+                'generateNewMappings(lecture.videoId, images, duration)'
+            ],
+            _videoUrlChanged: function(videoUrl) {
+                if (videoUrl === undefined || videoUrl === "") {
+                    return;
+                }
+                var videoId = UrlUtils.getParam(videoUrl, "v");
+                this.set('lecture.videoId', videoId);
+            },
+            _updateSelectedSection: function(e) {
+                var self = this;
+
+                function getNextTime(index) {
+                    if (index < self.lecture.mappings.length - 1) {
+                        return self.lecture.mappings[index + 1].time;
+                    }
+                    return self.duration;
                 }
+
+                this.selectedIndex = e.detail.index
+
+                console.log("Updating form");
+                this.timeFrom = this.lecture.mappings[this.selectedIndex].time.toHHMMSS();
+                this.timeTo = getNextTime(this.selectedIndex).toHHMMSS();
+
+                var isFirst = this.selectedIndex === 0;
+                this.$.fromButton.disabled = isFirst;
+                this.$.fromInput.disabled = isFirst;
+
+                var isLast = this.selectedIndex === this.lecture.mappings.length - 1;
+                this.$.toButton.disabled = isLast;
+                this.$.toInput.disabled = isLast;
+
+                console.log("Updating video and presentation view");
+                var selectedMapping = this.lecture.mappings[this.selectedIndex];
+                this.async(function() {
+                    this.$.presentation.seekTo(selectedMapping.time)
+                }, 10);
+                this.$.imgDisplay.set("img", this.images[selectedMapping.slide]);
             },
-            _updateSelectedTime: function (e) {
-                this.$.mapSlide.time = e.detail.value;
+            _youtubeReady: function(e) {
+                this.set('duration', e.detail.target.F.duration);
             },
-            _updateSelectedSlide: function (e) {
-                this.$.mapSlide.slide = e.detail;
+            _youtubeStateChanged: function(e) {
+                var newDuration = Math.round(e.detail.target.F.duration);
+                if (newDuration !== 0 && this.duration !== newDuration) {
+                    this.set('duration', e.detail.target.F.duration);
```

Comment:
e.detail.target.F.duration - to się ciągle powtarzaa, przenieś do jakiejś funkcji żeby było w jednym miejscu.